text/0097-log-data-model.md

```diff
@@ -0,0 +1,988 @@
+    "of": {
+      "some": "complexity"
+    }
+  }
+}
+```
+
+Example 3
+
+```javascript
+{
+   "Timestamp": 1586960586000,
+   "Attributes":{
+      "http.scheme":"https",
+      "http.host":"donut.mycie.com",
+      "http.target":"/order",
+      "http.method":"post",
+      "http.status_code":500,
+      "http.flavor":"1.1",
+      "http.user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36",
+   }
+}
+```
+
+## Open Questions
+
+- Should we store entire
+  [W3C Trace Context](https://www.w3.org/TR/trace-context/), including
+  `traceparent` and `tracestate` fields instead of only `TraceFlags`?
+  
+- Is `SeverityText`/`SeverityNumber` fields design good enough?
+
+- Early draft of this proposal specified that `Timestamp` should be populated
+  from a monotonic, NTP-synchronized source. I removed this requirement to avoid
+  confusion. Do we need any requirements for timestamp sources?
+
+- Is there a need for special treatment of security logs?
+
+## Alternate Design
+
+An
+[alternate design](https://docs.google.com/document/d/1ix9_4TQO3o-qyeyNhcOmqAc1MTyr-wnXxxsdWgCMn9c/edit?ts=5e990fe2#heading=h.cw69q2ga62p6)
+that used an envelop approach was considered but I did not find it to be overall
+better than this one.
+
+## Prior Art
+
+### RFC5424 Syslog
+
+[RFC5424](https://tools.ietf.org/html/rfc5424) defines structured log data
+format and protocol. The protocol is ubiquitous (although unfortunately many
+implementations don’t follow structured data recommendations). Here are some
+drawbacks that do not make Syslog a serious contender for a data model:
+
+- While it allows structured attributes the body of the message can be only a
+  string.
+
+- Severity is hard-coded to 8 possible numeric values, and does not allow custom
+  severity texts.
+
+- Structured data does not allow arbitrary nesting and is 2-level only.
+
+- No clear separate place to specify data source (aka resource). There are a
+  couple hard-coded fields that serve this purpose in a limited way (HOSTNAME,
+  APP-NAME, FACILITY).
+
+### Fluentd Forward Protocol Model
+
+[Forward protocol](https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1)
+defines a log Entry concept as a timestamped record. The record consists of 2
+elements: a tag and a map of arbitrary key/value pairs.
+
+The model is universal enough to represent any log record. However, here are
+some drawbacks:
+
+- All attributes of a record are represented via generic key/value pairs (except
+  tag and timestamp). This misses the optimization opportunities (see [Design
+  Notes](#design-notes)).
+
+- There is no clear separate place to specify data source (aka resource).
+
+- There is no mention of how exactly keys should be named and what are expected
+  values. This lack of any naming convention or standardization of key/value
+  pairs makes interoperability difficult.
+
+## Appendix A. Example Mappings
```

Comment:
Would be cool to see a section for mapping from [Elastic Common Schema](https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html), which is another modern effort to provide a common data model for events (with a larger scope, since it involves more than just log events)

Reply:
It is work-in-progress by @roncohen at https://docs.google.com/document/d/1ix9_4TQO3o-qyeyNhcOmqAc1MTyr-wnXxxsdWgCMn9c/edit?ts=5e985b59#heading=h.er24d9gosi03
Once finalized we can add to this PR.